src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -703,4 +707,479 @@ private async Task<HttpResponseMessage> GetVersion(Guid dataSetId, string dataSe
+                            var locationSchoolOptionMetaViewModel = 
+                                Assert.IsType<LocationSchoolOptionMetaViewModel>(locationOptionMetaViewModel);
+                            Assert.Equal(
+                                locationSchoolOptionMeta.Label, 
+                                locationSchoolOptionMetaViewModel.Label);
+                            Assert.Equal(
+                                locationSchoolOptionMeta.Urn, 
+                                locationSchoolOptionMetaViewModel.Urn);
+                            Assert.Equal(
+                                locationSchoolOptionMeta.LaEstab, 
+                                locationSchoolOptionMetaViewModel.LaEstab);
+                            break;
+                        default:
+                            throw new NotImplementedException();
+                    }
+                }
+            }
+
+            Assert.Equal(dataSetVersion.LocationMetas.Select(lm => lm.Level), content.GeographicLevels);
+
+            foreach (var indicator in content.Indicators)
+            {
+                var indicatorMeta = Assert.Single(dataSetVersion.IndicatorMetas, im => im.PublicId == indicator.Id);
+
+                Assert.Equal(indicatorMeta.Label, indicator.Label);
+                Assert.Equal(indicatorMeta.Unit, indicator.Unit);
+                Assert.Equal(indicatorMeta.DecimalPlaces, indicator.DecimalPlaces);
+            }
+
+            foreach (var timePeriod in content.TimePeriods)
+            {
+                var timePeriodMeta = Assert.Single(
+                    dataSetVersion.TimePeriodMetas, 
+                    tp => tp.Code == timePeriod.Code
+                    && tp.Period == timePeriod.Period);
+
+                Assert.Equal(
+                    TimePeriodFormatter.FormatLabel(timePeriodMeta.Period, timePeriodMeta.Code), 
+                    timePeriod.Label);
+            }
+        }
+
+        [Fact]
+        public async Task VersionSpecified_ReturnsCorrectVersion()
+        {
+            DataSet dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatusPublished();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            var dataSetVersions = DataFixture
+                .DefaultDataSetVersion(
+                    filters: 1,
+                    indicators: 1,
+                    locations: 1,
+                    timePeriods: 3)
+                .WithStatusPublished()
+                .WithDataSetId(dataSet.Id)
+                .ForIndex(0, dsv => dsv.SetVersionNumber(1, 0))
+                .ForIndex(0, dsv => dsv.SetFilterMetas(() =>
+                    [
+                        DataFixture
+                        .DefaultFilterMeta()
+                        .WithLabel("filter 1")
+                    ]
+                ))
```

Comment:
Don't have to `ForIndex` twice. Can roll this up into a single one i.e.

```cs
.ForIndex(0, dsv => dsv
    .SetVersionNumber(1, 0)
    .SetFilterMetas(() =>
    [
        DataFixture
            .DefaultFilterMeta()
            .WithLabel("filter 1")
    ]))
```

Similar thing applies for the other indices below